lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -325,7 +327,17 @@ public void finalize() {
 	 * @throws AblyException
 	 */
 	<T> T ablyHttpExecute(String path, String method, Param[] headers, Param[] params, RequestBody requestBody, ResponseHandler<T> responseHandler) throws AblyException {
-		int retryCountRemaining = Hosts.isRestFallbackSupported(this.host) ? options.httpMaxRetryCount : 0;
+		List<String> customFallbackHosts = null;
+		int retryCountRemaining = 0;
+		if (options.fallbackHosts != null) {
+			if (options.fallbackHosts.length > 0) {
```

Comment:
If ClientOptions#fallbackHosts array isn't empty, we should use only these fallback hosts. So, if user want to disable fallback functionality, user should provide specific restHost (it isn't default primary host) or empty array of ClientOptions#fallbackHosts.